bundles/ch.elexis.core.services/src/ch/elexis/core/services/AppointmentService.java

```diff
@@ -876,4 +878,106 @@ private boolean isOverlapping(LocalDateTime start1, LocalDateTime end1, LocalDat
 		}
 		return false;
 	}
+
+	@Override
+	public List<IAppointment> createKombiTermineIfApplicable(IAppointment mainAppointment, IContact patient,
+			String type, String patientName) {
+		List<IAppointment> kombiAppointments = new ArrayList<>();
+		List<String> kombiTermineList = ConfigServiceHolder.get().getAsList(AG_KOMBITERMINE + type);
+		if (kombiTermineList.isEmpty()) {
+			return Collections.emptyList();
+		}
+
+		if (!AppointmentExtensionHandler.getLinkedAppointments(mainAppointment).isEmpty()) {
+			return Collections.emptyList();
+		}
+		AppointmentExtensionHandler.setMainAppointmentId(mainAppointment, mainAppointment.getId());
+		List<String> linkedIds = new ArrayList<>();
+
+		for (String kombiTermin : kombiTermineList) {
+			kombiTermin = kombiTermin.replaceAll("[{}]", "");
+			String[] elements = kombiTermin.split(";");
+			if (elements.length < 6)
+				continue;
+
+			IAppointment newAppointment = CoreModelServiceHolder.get().create(IAppointment.class);
+			newAppointment.setState(mainAppointment.getState());
+			newAppointment.setType(elements[2]);
+			newAppointment.setSchedule(elements[1]);
+			newAppointment.setCreatedBy(mainAppointment.getCreatedBy());
+			newAppointment.setCreated(createTimeStamp());
+			newAppointment.setLastEdit(createTimeStamp());
+			newAppointment.setReason(elements[0]);
+
+			if (patient != null) {
+				newAppointment.setSubjectOrPatient(patient.getId());
+			} else if (StringUtils.isNotBlank(patientName)) {
+				newAppointment.setSubjectOrPatient(patientName);
+			}
+
+			LocalDateTime startTime = mainAppointment.getStartTime();
+			int offset = Integer.parseInt(elements[4]);
+			if (Messages.AddCombiTerminDialogBefore.equalsIgnoreCase(elements[3])) {
+				startTime = startTime.minusMinutes(offset);
+			} else {
+				startTime = startTime.plusMinutes(offset);
+			}
+			newAppointment.setStartTime(startTime);
+			newAppointment.setEndTime(startTime.plusMinutes(Integer.parseInt(elements[5])));
+			AppointmentExtensionHandler.setMainAppointmentId(newAppointment, mainAppointment.getId());
+			AppointmentExtensionHandler.addLinkedAppointmentId(newAppointment, newAppointment.getId());
+			linkedIds.add(newAppointment.getId());
+
+			kombiAppointments.add(newAppointment);
+		}
+		AppointmentExtensionHandler.addMultipleLinkedAppointments(mainAppointment, linkedIds);
+		return kombiAppointments;
+	}
+
+	@Override
+	public List<IAppointment> findCollisionsForKombiAppointment(IAppointment newAppointment, String appointmentType) {
```

Comment:
Bitte in `isColliding` umbenennen. Und dann nur 1 Parameter `List<IAppointments> appointments`.

Die Methode soll nur die Liste von Appointments auf Kollisionen prüfen, und die mit Kollisionen zurück geben.